plugins/bigquery/dbt/include/bigquery/macros/catalog.sql

```diff
@@ -1,8 +1,13 @@
 
 {% macro bigquery__get_catalog(information_schemas) -%}
 
-  {%- call statement('catalog', fetch_result=True) -%}
-    {% for information_schema in information_schemas %}
+  {%- if (information_schemas | length) == 0 -%}
+    {# Hopefully nothing cares about the columns we return when there are no rows #}
+    {%- set query  = "select 1 as id limit 0" -%}
```

Comment:
Can we `return(none)` here and make the `get_catalog` method error with a sensible message?

Reply:
Should it really error instead of the catalog just being empty? As it is this works, it just generates an empty catalog.

Reply:
oh, ok, that's great. Ship this as-is :D